src/main/scala/com/fulcrumgenomics/alignment/Alignment.scala

```diff
@@ -40,6 +42,34 @@ case class CigarElem(operator: CigarOperator, length: Int) {
   override def toString: String = length + operator.toString
 }
 
+/** Companion object for Cigar that offers alternative constructors. */
+object Cigar {
+  private val ZeroCharAsInt = '0'.toInt
+
+  /** Constructs a Cigar object from a Cigar string. */
+  def apply(cigar: String): Cigar = {
+    require(cigar != null && cigar.length > 0, "Can't parse a null or empty cigar string.")
```

Comment:
Or you could let it blow up in the cigar constructor for an empty string.  Probably nice to have it here so the error message is more specific.